### PR TITLE
Update C API to allow user to enable caching of feeds and fetches info across calls to Run

### PR DESCRIPTION
**Pull request by skottmckay**
Add ability to enable caching to the C API, and update the internals to pass the feed names and MLValue instances in vectors so the order is deterministic (so cache entry matching works as expected).

### CODE REVIEW

File: include/onnxruntime/core/session/onnxruntime_c_api.h [hunk 1]
```diff
@@ -261,9 +261,11 @@ ORT_API(OrtRunOptions*, OrtCreateRunOptions);
 
 ORT_API_STATUS(OrtRunOptionsSetRunLogVerbosityLevel, _In_ OrtRunOptions*, unsigned int);
 ORT_API_STATUS(OrtRunOptionsSetRunTag, _In_ OrtRunOptions*, _In_ const char* run_tag);
+ORT_API(void, OrtRunOptionsSetCacheFeedsFetchesInfoEnabled, _In_ OrtRunOptions* options, bool value);
```
Comment (pranavsharma) [inline include/onnxruntime/core/session/onnxruntime_c_api.h]: OrtRunOptionsSetTerminate uses an 'int' to set a boolean flag. We should be consistent with the usage of boolean types. Are we using int or bool?

Comment (skottmckay) [reply include/onnxruntime/core/session/onnxruntime_c_api.h]: 'int' is probably cleaner.

**Comment by pranavsharma** [comment]
LGTM 👍

**Comment by pranavsharma** [comment]
As an aside, I'm wondering if we've to have a new function for every single option we introduce. Well known APIs like setsockopt, curl_easy_opt do it differently. You've only one API to set the options and just use an enum for the various options.

File: onnxruntime/test/shared_lib/test_run_options.cc [hunk 2]
```diff
@@ -12,7 +12,7 @@ TEST_F(CApiTest, run_options) {
   ASSERT_EQ(OrtRunOptionsSetRunTag(options.get(), "abc"), nullptr);
   ASSERT_STREQ(OrtRunOptionsGetRunTag(options.get()), "abc");
   ASSERT_EQ(OrtRunOptionsGetRunLogVerbosityLevel(options.get()), (unsigned)1);
-  ASSERT_EQ(OrtRunOptionsGetCacheFeedsFetchesInfoEnabled(options.get()), false);
-  OrtRunOptionsSetCacheFeedsFetchesInfoEnabled(options.get(), true);
-  ASSERT_EQ(OrtRunOptionsGetCacheFeedsFetchesInfoEnabled(options.get()), true);
+  ASSERT_EQ(OrtRunOptionsGetCacheFeedsFetchesInfoEnabled(options.get()), int(0));
```
Comment (snnn) [inline onnxruntime/test/shared_lib/test_run_options.cc]: Just a coding style suggestion: you can either write static_cast<int>(0), or int(0), or (int)0.  But if we mix them in one file, even put them in adjacent lines, it's not beautiful.

**Pull request merged**